src/dfm/data/preprocess.py

```diff
@@ -0,0 +1,81 @@
+"""Data preprocessing script for Danish Foundation Models """
+from typing import List, Union
+from functools import partial
+from datasets.arrow_dataset import Dataset
+from transformers import AutoTokenizer
+from transformers.tokenization_utils_base import PreTrainedTokenizerBase
+from transformers.tokenization_utils_fast import PreTrainedTokenizerFast
+from dfm.data.load import dfm_load_dataset
+from datasets import DatasetDict
+
+
+def main():
+    tokenizer = AutoTokenizer.from_pretrained("Maltehb/danish-bert-botxo")
+
+    ds = dfm_load_dataset("DDSC/reddit-da")
+    ds["train"] = ds["train"].select(range(1000))
+    ds["test"] = ds["test"].select(range(1000))
+    ds["val"] = ds["val"].select(range(1000))
+
+    ds = preprocess_dataset(ds, tokenizer)
+
+
+def preprocess_dataset(
+    dataset: DatasetDict,
+    tokenizer: Union[PreTrainedTokenizerFast, PreTrainedTokenizerBase],
+    num_proc: int = 4,
+    block_size: int = 512,
+):
+
+    for key in dataset.keys():
+        cols = dataset[key].column_names
+        cols.remove("text")
+        dataset[key] = dataset[key].remove_columns(cols)
+        # select the text column
+        ##dataset[key] = dataset[key].map(
+        #    select_column_, input_columns="text", remove_columns=dataset[key].column_names
+        # )
+
+    tokenize_func_ = partial(tokenize_func, tokenizer=tokenizer)
+    dataset = dataset.map(
+        tokenize_func_, batched=True, num_proc=num_proc, remove_columns=["text"]
+    )
+    group_texts_ = partial(group_texts, block_size=block_size)
+
+    dataset = dataset.map(
+        group_texts_,
+        batched=True,
+        batch_size=1000,
+        num_proc=num_proc,
+    )
+    dataset.shuffle()
```

Comment:
This doesn't perform a proper shuffle as `dataset` is an iterable dataset, right? Maybe make that clear if it's the case.